tests/System.Text.Primitives.Tests/PrimitivesPerfTests.cs

```diff
@@ -0,0 +1,875 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using Xunit;
+using Microsoft.Xunit.Performance;
+
+namespace System.Text.Primitives.Tests
+{
+    public class PrimitivesPerfTests
+    {
+        private static byte[] UtfEncode(string s)
+        {
+            return Encoding.UTF8.GetBytes(s);
+        }
+
+        #region byte
+        [Benchmark]
+        [InlineData("128")] // standard parse
+        [InlineData("300")] // basic overflow
+        [InlineData("1281995")] // heavy overflow
+        [InlineData("0128")] // leading zero
+        [InlineData("00000000128")] // many leading zeroes
+        [InlineData("-128")] // negative value handling for unsigned types
+        [InlineData("255")] // Max value
+        [InlineData("0")] // Min value
+        private static void BaselineByteParse(string text)
+        {
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                byte value;
+                using (iteration.StartMeasurement())
+                {
+                    Byte.TryParse(text, NumberStyles.None, CultureInfo.InvariantCulture, out value);
```

Comment:
This work is too inexpensive. I worry that the cost of the framework calls is a big par of it. Would it make sense to parse this multiple times per iteration?